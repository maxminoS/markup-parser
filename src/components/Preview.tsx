import { parseMarkdown, parseOrg } from "utils/Parsers";

type Props = {
  parseTo: string,
  content: string,
}

export const Preview = (props: Props) => {
  return (
      <div style={{
            border: "1px solid black",
            display: "inline-block",
            width: "96%",
            height: "85vh",
            overflowY: "auto"
          }}
           dangerouslySetInnerHTML={{
             __html: props.parseTo === "Markdown" ? parseMarkdown(props.content) :
                     props.parseTo === "Org Mode" ? parseOrg(props.content) : props.content
           }}
      >
      </div>
  );
}
