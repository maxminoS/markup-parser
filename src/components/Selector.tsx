import React from 'react';
import { FormControl, InputLabel, Select, MenuItem } from "@material-ui/core";

type Props = {
  parseTo: string,
  setParseTo: (parseType: string) => void,
}

export const Selector = (props: Props) => {
  const handleSelectChange = (e: React.ChangeEvent<{ value: unknown }>) => props.setParseTo(e.target.value as string);

  return (
    <FormControl variant="outlined">
      <InputLabel>Markup</InputLabel>
      <Select
        value={props.parseTo}
        onChange={handleSelectChange}
        label="Markup"
      >
        <MenuItem value="HTML">HTML</MenuItem>
        <MenuItem value="Markdown">Markdown</MenuItem>
        <MenuItem value="Org Mode">Org Mode</MenuItem>
      </Select>
    </FormControl>
  );
}
