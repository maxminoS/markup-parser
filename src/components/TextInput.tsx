import React from 'react';

type Props = {
  content: string,
  setContent: (content: string) => void,
}

export const TextInput = (props: Props) => {
  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => props.setContent(e.target.value);

  return(
    <textarea
          onChange={handleChange}
          style={{
              width: "98%",
              height: "85vh"
          }}
          value={props.content}
    />
  );
}
