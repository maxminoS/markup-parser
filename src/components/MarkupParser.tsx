import React, { useState } from "react";
import { AppBar, Toolbar, Grid, Box, Typography } from "@material-ui/core";

import { Preview } from "components/Preview";
import { TextInput } from "components/TextInput";
import { Selector } from "components/Selector";

export const MarkupParser = () => {
  const [content, setContent] = useState("");
  const [parseTo, setParseTo] = useState("HTML");

  return (
    <>
      <AppBar position="static" color="primary">
        <Toolbar style={{ minHeight: "75px" }}>
          <Box display="flex" flexGrow={1}>
            <Typography variant="h6" align="center">Markup Parser</Typography>
          </Box>
          <Selector parseTo={parseTo} setParseTo={setParseTo} />
        </Toolbar>
      </AppBar>
      <br />
      <Grid container
            justify="center"
            alignItems="flex-start"
            style={{ minHeight: "88vh" }}
            spacing={2}
      >
        <Grid item xs={5}>
          <TextInput content={content} setContent={setContent} />
        </Grid>
        <Grid item xs={5}>
          <Preview parseTo={parseTo} content={content} />
        </Grid>
      </Grid>
    </>
  );
}
