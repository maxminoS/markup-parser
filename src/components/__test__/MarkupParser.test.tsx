import React from 'react';
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

import { MarkupParser } from "components/MarkupParser";

describe('Previews', () => {
  beforeEach(() => {
    render(<MarkupParser />);
  });

  it('previews HTML correctly', () => {
    userEvent.type(screen.getByRole("textbox"), "<h1>Preview this text</h1>");
    expect(screen.getByRole("heading", { name: /preview this text/i })).toBeInTheDocument();
  });

  it('previews Markdown correctly', () => {
    userEvent.click(screen.getByRole("button", { name: /html/i }));
    userEvent.click(screen.getByRole("option", { name: /markdown/i }));

    userEvent.type(screen.getByRole("textbox"), "# Preview this text");
    expect(screen.getByRole("heading", { name: /preview this text/i })).toBeInTheDocument();
  });

  it('previews Org Mode correctly', () => {
    userEvent.click(screen.getByRole("button", { name: /html/i }));
    userEvent.click(screen.getByRole("option", { name: /org mode/i }));

    userEvent.type(screen.getByRole("textbox"), "* Preview this text");
    expect(screen.getByRole("heading", { name: /preview this text/i })).toBeInTheDocument();
  });
});

it('renders MarkupParser component as expected', () => {
  const markupParser = renderer.create(<MarkupParser />).toJSON();
  expect(markupParser).toMatchSnapshot();
});
