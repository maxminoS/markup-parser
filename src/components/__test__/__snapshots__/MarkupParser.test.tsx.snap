// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MarkupParser component as expected 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      style={
        Object {
          "minHeight": "75px",
        }
      }
    >
      <div
        className="MuiBox-root MuiBox-root-16"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
        >
          Markup Parser
        </h6>
      </div>
      <div
        className="MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink={true}
        >
          Markup
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            HTML
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="HTML"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
            >
              <span>
                Markup
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </header>,
  <br />,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
    style={
      Object {
        "minHeight": "88vh",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
    >
      <textarea
        onChange={[Function]}
        style={
          Object {
            "height": "85vh",
            "width": "98%",
          }
        }
        value=""
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
        style={
          Object {
            "border": "1px solid black",
            "display": "inline-block",
            "height": "85vh",
            "overflowY": "auto",
            "width": "96%",
          }
        }
      />
    </div>
  </div>,
]
`;
